pkg/fs/lock.go

```diff
@@ -0,0 +1,66 @@
+package fs
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const lockFilename = "LOCK"
+
+// ClaimLock on the specified path.
+// Successful claims must eventually be released.
+func ClaimLock(fs Filesystem, path string) error {
+	lockFile := filepath.Join(path, lockFilename)
+	if fs.Exists(lockFile) {
```

Comment:
Oh yes, that would make me feel much better. :) Was going to say too that this kind of locking is racy in general.